pydicom/tests/test_dataset.py

```diff
@@ -1411,6 +1411,1037 @@ def test_update_with_dataset(self):
+    },
+    "00180023": {
+        "vr": "CS",
+        "Value": [
+            "2D"
+        ]
+    },
+    "00180022": {
+        "vr": "CS",
+        "Value": [
+            "EDR_GEMS",
+            "EPI_GEMS"
+        ]
+    },
+    "00180025": {
+        "vr": "CS",
+        "Value": [
+            "N"
+        ]
+    },
+    "00209057": {
+        "vr": "UL",
+        "Value": [
+            13
+        ]
+    },
+    "00200011": {
+        "vr": "IS",
+        "Value": [
+            10
+        ]
+    },
+    "00200010": {
+        "vr": "SH",
+        "Value": [
+            ""
+        ]
+    },
+    "00280004": {
+        "vr": "CS",
+        "Value": [
+            "MONOCHROME2"
+        ]
+    },
+    "00200012": {
+        "vr": "IS",
+        "Value": [
+            1
+        ]
+    },
+    "00280002": {
+        "vr": "US",
+        "Value": [
+            1
+        ]
+    },
+    "00080060": {
+        "vr": "CS",
+        "Value": [
+            "MR"
+        ]
+    },
+    "00081111": {
+        "vr": "SQ",
+        "Value": [
+            {
+                "00081150": {
+                    "vr": "UI",
+                    "Value": [
+                        "1.2.840.10008.3.1.2.3.3"
+                    ]
+                },
+                "00081155": {
+                    "vr": "UI",
+                    "Value": [
+                        "1.3.6.1.4.1.5962.99.1.1516012269.2092621929.1509049565933.1718.0"
+                    ]
+                }
+            }
+        ]
+    },
+    "00080008": {
+        "vr": "CS",
+        "Value": [
+            "ORIGINAL",
+            "PRIMARY",
+            "OTHER"
+        ]
+    },
+    "00180050": {
+        "vr": "DS",
+        "Value": [
+            4.0
+        ]
+    },
+    "00200013": {
+        "vr": "IS",
+        "Value": [
+            151
+        ]
+    },
+    "00101010": {
+        "vr": "AS",
+        "Value": [
+            "051Y"
+        ]
+    },
+    "00251007": {
+        "vr": "SL",
+        "Value": [
+            714
+        ]
+    },
+    "00080080": {
+        "vr": "LO",
+        "Value": [
+            "Brigham and Womens Hospital"
+        ]
+    },
+    "00280010": {
+        "vr": "US",
+        "Value": [
+            256
+        ]
+    },
+    "00181100": {
+        "vr": "DS",
+        "Value": [
+            180.0
+        ]
+    },
+    "00080032": {
+        "vr": "TM",
+        "Value": [
+            "134345"
+        ]
+    },
+    "00200060": {
+        "vr": "CS",
+        "Value": [
+            ""
+        ]
+    },
+    "00080070": {
+        "vr": "LO",
+        "Value": [
+            "GE MEDICAL SYSTEMS"
+        ]
+    }
+}
+"""
+        ds = Dataset.from_json(jsonmodel)
+        import json
+        jsonmodel2 = ds.to_json(dump_handler=lambda d: json.dumps(d, indent=2))
+        ds2 = Dataset.from_json(jsonmodel2)
+
+        assert ds.PatientIdentityRemoved == 'YES'
+        assert ds2.PatientIdentityRemoved == 'YES'
 
```

Comment:
Are there any tests to confirm values within sequences?  Here a round-trip is done, but I don't see any comparison of the two datasets to each other except indirectly that both can return PatientIdentityRemoved.